# HG changeset patch
# User Rik <rik@octave.org>
# Date 1391408505 28800
#      Sun Feb 02 22:21:45 2014 -0800
# Node ID cb377af34c00e65ac7a01fbe94399bdd72251d70
# Parent  1597281144e4cbaa3956bbe82659183b112c4716
qqplot.m: Plot points as well as connecting line.

* qqplot.m: Plot points as well as connecting line.

diff --git a/scripts/statistics/base/qqplot.m b/scripts/statistics/base/qqplot.m
--- a/scripts/statistics/base/qqplot.m
+++ b/scripts/statistics/base/qqplot.m
@@ -57,17 +57,17 @@
 ## Description: Perform a QQ-plot (quantile plot)
 
 function [qout, sout] = qqplot (x, dist, varargin)
 
   if (nargin < 1)
     print_usage ();
   endif
 
-  if (!(isnumeric (x) && isvector (x)))
+  if (! (isnumeric (x) && isvector (x)))
     error ("qqplot: X must be a numeric vector");
   endif
 
   if (nargin == 1)
     f = @stdnormal_inv;
   else
     if (isnumeric (dist))
       f = @(y) empirical_inv (y, dist);
@@ -92,23 +92,22 @@ function [qout, sout] = qqplot (x, dist,
     else
       q_label = sprintf ("%s with parameters %g", func2str (f), varargin{1});
       param_str = sprintf (", %g", varargin{2:end});
       q_label = [q_label param_str]; 
     endif
   endif
 
   if (nargout == 0)
-    plot (q, s);
+    plot (q, s, "-x");
     q_label = strrep (q_label, '_inv', '\_inv');
     if (q_label(1) == '@')
       q_label = q_label(6:end);  # Strip "@(y) " from anon. function
     endif
     xlabel (q_label);
     ylabel ("sample points");
   else
     qout = q;
     sout = s;
   endif
 
 endfunction
 
-
